libcodechecker/store.py

```diff
@@ -308,6 +314,25 @@ def __get_run_name(input_list):
         return False
 
 
+def process_compile_db(compile_db):
+    """
+    Create a dict where the source file name (absolute path) is the key and
+    the correspoding compilation command is the value.
+    """
+    compile_db = os.path.abspath(compile_db)
+
+    comp_cmds = {}
+    with open(compile_db, 'r') as ccdb:
+        comp_cmds = json.load(ccdb)
+
+    processed = {}
+    for cc in comp_cmds:
+        processed[str(os.path.join(cc['directory'], cc['file']))] = \
+                str(' '.join(cc['arguments']))
```

Comment:
Is this compilation database the one which is created by `CodeChecker log`? If I run `CodeChecker log` then the resulting JSON file has no "arguments" attribute but "command". And the "file" attribute contains the full path of the file, not just the file name. However I have seen a compilation database which contained a relative path to the file. It was generated by scan-build if I remember well.